[DOCS] Refactor glossary (#70516)
elastic/elasticsearch: docs/reference/glossary.asciidoc

// end::field-def[]

[[glossary-filter]] filter::
// tag::filter-def[]
<<glossary-query,Query>> that does not score matching documents. See
{ref}/query-filter-context.html[filter context].
// end::filter-def[]

[[glossary-flush]] flush::
// tag::flush-def[]
Writes data from the {ref}/index-modules-translog.html[transaction log] to disk
for permanent storage. See the {ref}/indices-flush.html[flush API].
// end::flush-def[]

[[glossary-follower-index]] follower index::
// tag::follower-index-def[]
Target <<glossary-index,index>> for <<glossary-ccr,{ccr}>>. A follower index
exists in a <<glossary-local-cluster,local cluster>> and replicates a
<<glossary-leader-index,leader index>>. See {ref}/xpack-ccr.html[{ccr-cap}].
// end::follower-index-def[]

[[glossary-force-merge]] force merge::
// tag::force-merge-def[]
// tag::force-merge-def-short[]
Manually triggers a <<glossary-merge,merge>> to reduce the number of
<<glossary-segment,segments>> in an index's <<glossary-shard,shards>>.
// end::force-merge-def-short[]
See the {ref}/indices-forcemerge.html[force merge API].
// end::force-merge-def[]

[[glossary-freeze]] freeze::
// tag::freeze-def[]
// tag::freeze-def-short[]

> Makes an index read-only and minimize its memory footprint.

Reviewer: ```suggestion
Makes an index read-only and minimizes its memory footprint.
```

See the {ref}/freeze-index-api.html[freeze API].
// end::freeze-def[]

[[glossary-frozen-index]] frozen index::
// tag::frozen-index-def[]
An <<glossary-index,index>> reduced to a low overhead state that still enables
occasional searches.
See the {ref}/freeze-index-api.html[freeze API].
// end::frozen-index-def[]

[[glossary-frozen-phase]] frozen phase::
// tag::frozen-phase-def[]
Fourth possible phase in the <<glossary-index-lifecycle,index lifecycle>>. In
the frozen phase, an <<glossary-index,index>> is no longer updated and
<<glossary-query,queried>> rarely. The information still needs to be searchable,
but it’s okay if those queries are extremely slow. See
{ref}/ilm-index-lifecycle.html[Index lifecycle].
// end::frozen-phase-def[]

[[glossary-frozen-tier]] frozen tier::
// tag::frozen-tier-def[]
<<glossary-data-tier,Data tier>> that contains <<glossary-node,nodes>> that
hold time series data that is accessed rarely and not normally updated. See
{ref}/data-tiers.html[Data tiers].
// end::frozen-tier-def[]

[[glossary-hidden-index]] hidden data stream or index::
// tag::hidden-index-def[]
<<glossary-data-stream,Data stream>> or <<glossary-index,index>> excluded from

> most <glossary-index-pattern,index patterns>> by default. See

Reviewer: ```suggestion
most <<glossary-index-pattern,index patterns>> by default. See
```

{ref}/multi-index.html#hidden-indices[Hidden data streams and indices].
// end::hidden-index-def[]

[[glossary-hot-phase]] hot phase::
// tag::hot-phase-def[]
First possible phase in the <<glossary-index-lifecycle,index lifecycle>>. In
the hot phase, an <<glossary-index,index>> is actively updated and queried. See
{ref}/ilm-index-lifecycle.html[Index lifecycle].
// end::hot-phase-def[]

[[glossary-hot-tier]] hot tier::
// tag::hot-tier-def[]

> <<glossary-data-tier,Data tier>> that contains <<glossary-node,nodes>> that
> handle the <<glossary-index,indexing>> load for time series data such as logs or
> metrics and hold your most recent, most-frequently-accessed data. See

Reviewer: ```suggestion
<<glossary-data-tier,Data tier>> that contains <<glossary-node,nodes>> that
handle the <<glossary-index,indexing>> load for time series data, such as logs or
metrics. This tier holds your most recent, most-frequently-accessed data. See
```

See {ref}/scalability.html[Clusters, nodes, and shards].
// end::node-def[]

[[glossary-primary-shard]] primary shard::
// tag::primary-shard-def[]
Lucene instance containing some or all data for an {es} index. When you index a
<<glossary-document,document>>, {es} first adds the document to the primary
shard before <<glossary-replica-shard,replica shards>>. See
{ref}/scalability.html[Clusters, nodes, and shards].
// end::primary-shard-def[]

[[glossary-query]] query::
// tag::query-def[]
Request for information about your data. You can think of a query as a
question, written in a way {es} understands. See
{ref}/search-your-data.html[Search your data].
// end::query-def[]

[[glossary-recovery]] recovery::
// tag::recovery-def[]
Process of syncing a <<glossary-replica-shard,replica shard>> from a
<<glossary-primary-shard,primary shard>>. Upon completion, the replica shard is
available for searches. See the {ref}/indices-recovery.html[index recovery API].
// end::recovery-def[]

[[glossary-reindex]] reindex::
// tag::reindex-def[]
Copies documents from a source to a destination. The source and destination can

> be an <<glossary-data-stream,data stream>>, <<glossary-index,index>>, or

Reviewer: ```suggestion
be a <<glossary-data-stream,data stream>>, <<glossary-index,index>>, or
```

repository can be a shared filesystem or a remote repository, such as Azure or
Google Cloud Storage. See {ref}/snapshot-restore.html[Snapshot and restore].
// end::snapshot-repository-def[]

[[glossary-source_field]] source field::
// tag::source-field-def[]
Original JSON object provided during <<glossary-index,indexing>>. See the
{ref}/mapping-source-field.html[`_source` field].
// end::source-field-def[]

[[glossary-split]] split::
// tag::split-def[]
Adds more <<glossary-primary-shard,primary shards>> to an
<<glossary-index,index>>. See the {ref}/indices-split-index.html[split index
API].
// end::split-def[]

[[glossary-system-index]] system index::
// tag::system-index-def[]
<<glossary-index,Index>> containing configurations and other data used
internally by the {stack}. System index names start with a dot (`.`), such as
`.security`. Do not directly access or change system indices.
// end::system-index-def[]

[[glossary-term]] term::
// tag::term-def[]
See {ref}/glossary.html#glossary-token[token].
// end::term-def[]

[[glossary-text]] text::
// tag::text-def[]
Unstructured content, such as a product description or log message. You

> typically <<glossary-analysis,analyze>> text is for better search. See

Reviewer: ```suggestion
typically <<glossary-analysis,analyze>> text for better search. See
```
